doc/compiler/aot/LoadingByDependency.md

```diff
@@ -0,0 +1,108 @@
+
+## Tracking dependencies for AOT compilations
+
+To have the dependency information for a cached method readily available, the relocatable compilation infrastructure will use `J9::Compilation::addAOTMethodDependency()` to make a note of:
+
+1. The class chain offset of any class that will be relevant for relocation
+2. The initialization status of that class
+
+These dependencies are encoded in a "dependency chain" after the relocation records of the method are processed in `J9::AheadOfTimeCompile::processRelocations()`, and the chain will be stored in the local SCC. The compiled method will then be flagged as having its dependencies tracked, and an offset to the chain will be stored in the AOT method header. As an optimization, if a method has zero dependencies then no chain is stored in the SCC - we need the separate "tracks dependencies" flag to distinguish those methods from ones that do not track dependencies at all, because those latter methods must use traditional count-based loading.
+
+As it happens, the only class load/initialization dependencies for a compiled method that need to be tracked are ones for class chain offsets that are mentioned in one of the relocation records of that method. These are the only classes that the relo runtime will want to resolve during relocation. For that reason, the dependencies are mostly gathered in `RelocationRecord.cpp`, in the methods that set the class chain offsets that are stored in the relo records. Specifically, a dependency is registered right next to every call to `J9::AheadOfTimeCompile::addClassSerializationRecord` and `J9::AheadOfTimeCompile::addClassChainSerializationRecord`, because those methods are what register all the class/class chain offset uses for JITServer AOT cache compilations (and so must cover every such use). Some dependency tracking also takes place in `SymbolValidationManager.cpp`, mostly related to well-known classes; instead of adding the contents of the entire well-known classes chain as dependencies, we instead only add those well-known class chain offsets that are actually used in the relocation records. This reduces the number of dependencies somewhat.
```

Comment:
The second "instead" is not needed: "... instead of adding the contents of the entire well-known classes chain as dependencies, we **instead** only add ..."